wiki/Bazel-User-Guide.md

```diff
@@ -0,0 +1,277 @@
+
+## Running Tests
+
+The bazel test command is used to build and run tests in a Bazel workspace. It ensures that the specified test targets are built and executed in a sandboxed environment for reproducibility.
+
+**Syntax**
+
+```shell
+bazel test //path/to:target
+```
+
+### How to Obtain a Test Target
+A test target in Bazel refers to a specific testable entity defined in a BUILD file. It represents a test rule, such as `java_test` or `android_test`, that Bazel can build and execute.
+
+To ensure you always use the correct test target, follow one of these steps:
+1. If the test file is open in Android Studio:
+   * Right-click to open the context menu and select **Copy BUILD target string** as shown.
+     
+     ![Screenshot 2025-03-18 at 22 30 22](https://github.com/user-attachments/assets/286ebcb6-4f4e-4055-b6f0-4c070b039375)
+   
+   * Paste the target to the bazel test command, e.g. `bazel test //domain/src/test/java/org/oppia/android/domain/onboarding:AppStartupStateControllerTest`
+
+2. If the test file is open in Android Studio, but **Copy BUILD target string** is not available in the context menu:
+Sometimes the **Copy BUILD target string** is not available in the context menu either due to sync issues or modularization issues. You can still copy the file path, and pass it to the test command.
+   * In the AS sidebar, right click on the file, and select `copy path/reference`.
+     or
+   * Right-click on the file tab and then click `copy path/reference`.
+   * Next, select `copy path from repository root`. 
+   * Paste the copied path to the bazel test command in the terminal and remove the `.kt` extension, e.g. `bazel test domain/src/test/java/org/oppia/android/domain/onboarding/AppStartupStateControllerTest`.
+
+With this syntax, Bazel implicitly converts the file path into a target if it matches a test rule in the BUILD file. If the file is not part of a test target in the BUILD file, this command may fail.
+**When to use this**: If you are unsure about the exact Bazel target and want a quick way to run a test.
+
+3. Querying for a target
+Using the explicit Bazel target (// syntax) is more reliable compared to the file path syntax above. However, in cases such as running the test outside android studio, you may not know the target. In this case, you can use Bazel's `query` command to retrieve the correct test target:
+   * Copy the relative path of the test file.
+   * Run the following command in your terminal to get the test’s Bazel target: `bazel query relative-path-of-test-file`
+   Example
+   
+   ```shell
+   bazel query domain/src/test/java/org/oppia/android/domain/onboarding/AppStartupStateControllerTest.kt
+   ```
+   * The output will be a Bazel target that starts with //. Copy the target and remove the .kt extension.
+   * Use this target in the bazel test command, e.g. `bazel test //domain/src/test/java/org/oppia/android/domain/onboarding:AppStartupStateControllerTest`.
+
+### Common Flags
```

Comment:
Another good flag to add: `--runs_per_test` to test multiple runs: https://bazel.build/reference/command-line-reference#flag--runs_per_test.